testcases/misc_tests/p11sak_test.sh

```diff
@@ -0,0 +1,419 @@
+p11sak generate-key ec secp521r1 --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-ec-secp521r1
+
+
+echo "** Now list keys and redirect output to pre-files - 'p11sak_test.sh'"
+
+
+# list objects
+p11sak list-key des --slot $SLOT --pin $PKCS11_USER_PIN &> $P11SAK_DES_PRE
+p11sak list-key 3des --slot $SLOT --pin $PKCS11_USER_PIN &> $P11SAK_3DES_PRE
+p11sak list-key aes --slot $SLOT --pin $PKCS11_USER_PIN &> $P11SAK_AES_PRE
+p11sak list-key rsa --slot $SLOT --pin $PKCS11_USER_PIN &> $P11SAK_RSA_PRE
+p11sak list-key ec --slot $SLOT --pin $PKCS11_USER_PIN &> $P11SAK_EC_PRE
+
+
+echo "** Now remove keys - 'p11sak_test.sh'"
+
+
+# remove objects
+# des
+p11sak remove-key des --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-des -f
+# 3des
+p11sak remove-key 3des --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-3des -f
+# aes [128 | 192 | 256]
+p11sak remove-key aes --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-aes-128 -f
+p11sak remove-key aes --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-aes-192 -f
+p11sak remove-key aes --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-aes-256 -f
+# rsa [1024 | 2048 | 4096]
+# remove public key
+p11sak remove-key rsa --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-rsa-1024:pub -f
+p11sak remove-key rsa --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-rsa-2048:pub -f
+p11sak remove-key rsa --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-rsa-4096:pub -f
+# remove private key
+p11sak remove-key rsa --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-rsa-1024:prv -f
+p11sak remove-key rsa --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-rsa-2048:prv -f
+p11sak remove-key rsa --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-rsa-4096:prv -f
+# ec [prime256v1 | secp384r1 | secp521r1]
+#remove public key
+p11sak remove-key ec --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-ec-prime256v1:pub -f
+p11sak remove-key ec --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-ec-secp384r1:pub -f
+p11sak remove-key ec --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-ec-secp521r1:pub -f
+# remove private key
+p11sak remove-key ec --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-ec-prime256v1:prv -f
+p11sak remove-key ec --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-ec-secp384r1:prv -f
+p11sak remove-key ec --slot $SLOT --pin $PKCS11_USER_PIN --label p11sak-ec-secp521r1:prv -f
+
+
+echo "** Now list keys and rediirect to post-files - 'p11sak_test.sh'"
+
+
+# list objects
+p11sak list-key des --slot $SLOT --pin $PKCS11_USER_PIN &> $P11SAK_DES_POST
+p11sak list-key 3des --slot $SLOT --pin $PKCS11_USER_PIN &> $P11SAK_3DES_POST
+p11sak list-key aes --slot $SLOT --pin $PKCS11_USER_PIN &> $P11SAK_AES_POST
+p11sak list-key rsa --slot $SLOT --pin $PKCS11_USER_PIN &> $P11SAK_RSA_POST
+p11sak list-key ec --slot $SLOT --pin $PKCS11_USER_PIN &> $P11SAK_EC_POST
+
+
+echo "** Now checking output files to determine PASS/FAIL of tests - 'p11sak_test.sh'"
+
+
+# check DES
+grep -q 'p11sak-des' $P11SAK_DES_PRE
```

Comment:
OK, here you check if the list command shows the expected key. If the expected key is not there, then either the generate command has failed, or the list command has failed....
That's of course better than nothing, but still hard to distinguish what error has happened.

Reply:
That is a good point. Generally, the assumption is that the list command cannot display a key if the generate command failed. 
This is an initial test case that I want to expand to include also output from the eg. the p11tool. i. e. another tool generates the key and the p11sak tool lists the key. Also, this can then be used to actually check if an e. g. AES 192 key generated by the p11sak tool and the p11tool compare. This way, one can further build test cases that can pin point more specifically where the tool failed.